addons/account/models/account_account.py

```diff
@@ -386,36 +389,46 @@ def _inverse_code(self):
             # Need to set record.code with `company = self.env.company`, not `self.env.company.root_id`
             record_root.code_store = record.code
 
-    @api.depends_context('company')
+    @api.depends_context('allowed_company_ids')
     @api.depends('code')
     def _compute_placeholder_code(self):
         self.placeholder_code = False
         for record in self:
-            if record.code:
-                record.placeholder_code = record.code
-            elif authorized_companies := (record.company_ids & self.env['res.company'].browse(self.env.user._get_company_ids())).sorted('id'):
-                company = authorized_companies[0]
+            with contextlib.suppress(StopIteration):
+                company = next(c for c in self.env.companies if c.parent_ids & record.company_ids)
```

Comment:
Not a big fan of contextlib.suppress. Here, you could give None as default value for the call to next, and just add that to the condition below. Overall simpler and more readable IMO